test/inspector/test-async-hook-setup-at-signal.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common');
+common.skipIfInspectorDisabled();
+common.skipIf32Bits();
+const helper = require('./inspector-helper');
+const assert = require('assert');
+
+const script = `
+process._rawDebug('Waiting until a signal enables the inspector...');
+const waiting = setInterval(setupIntervalIfDebugged, 50);
+
+function setupIntervalIfDebugged() {
+ if (!process.binding('inspector').isEnabled()) return;
+ clearInterval(waiting);
+ // Introduce a small delay to allow V8 engine to correctly setup
+ // V8 Inspector async-task infrastructure.
+ setTimeout(setupIntervalWithBreak, 100);
```

Comment:
I know that timeouts make tests flaky, and also 100ms will be unnecessary long in most cases, which makes the test suite annoyingly slow.

Unfortunately I am not able to figure out what is causing this test to fail on slow machines. All I know is that somehow V8 Inspector is taking a bit longer to initialized AsyncTask tracking, even after `inspector()->enabled()` is already true. Maybe there are multiple threads involved? One handling the incoming SIGUSR1 signal, another setting up V8 Inspector IO thread, the third one running the actual Node.js event loop.

Can you advise what code to call in order to detect whether V8 Inspector has been fully initialized yet?